haystack/document_store/base.py

```diff
@@ -65,6 +66,20 @@ def get_all_documents(
         """
         pass
 
+    def __iter__(self):
+        if not self.ids_iterator:
+            self.ids_iterator = [x.id for x in self.get_all_documents()]
```

Comment:
I understand this is a generic solution in the BaseDocumentStore. Maybe it makes sense to implement a get_all_document_ids() method in the individual document stores (although it's more effort)? Or at least keep this in mind for future improvements? Otherwise this call will load a lot of data. 
In ElasticSearchDocumentStore, you could actually make use of the existing generator:
https://github.com/deepset-ai/haystack/blob/2a90471c730e8d5573a8766791563ebb7ac1cdff/haystack/document_store/elasticsearch.py#L601

Reply:
Yes I agree there are more efficient ways of implementing the Document Store as an iterator. I think that is outside the scope of this PR but we should think of reimplementing it, especially if this becomes an idiomatic way of accessing docs in the DocumentStore